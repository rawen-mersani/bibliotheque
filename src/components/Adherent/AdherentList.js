import React, {useState} from 'react';
import Adhrent from './Adherent.js'



export default function AdherentList(props) {

  const [isVisible, setIsVisible] = useState(true)
  

  const toggleVisibility=()=>{
    setIsVisible(!isVisible)
  }

 

  return (
    <div className="">
       <section className="content">
                        <div className="row">
                        <div className="col-xs-12">
                            <div className="box">
                            <div className="box-header">
                                <h3 className="box-title">Liste adherents</h3><br></br><br></br>
                                <button className="btn btn-info" onClick={toggleVisibility}>Afficher liste</button>
                            </div>

                            
                            {/* /.box-header */}
                            <div className="box-body">
                                <table id="example2" className="table table-bordered table-hover">
                                <thead>
                                    <tr>
                                      <th>Nom</th>
                                      <th>Prénom</th>
                                      <th>Mail</th>
                                      <th>Nombre emprunts</th>
                                      <th>Télephone</th>
                                      <th>Détails</th>
                                      <th>Modifier</th>
                                      <th>Supprimer</th>
                                    </tr>
                                </thead>
                                <tbody>
                                {
                                  
                                    isVisible && props.adherents.map( (adherent) => (
                                   <Adhrent key={adherent.id}
                                      id={adherent.id } 
                                      name={adherent.name} 
                                      pname ={adherent.pname} 
                                      mail ={adherent.mail} 
                                      nbEmp ={adherent.nbEmp}
                                      tel ={adherent.tel}
                                      deleteAdherent={props.deleteAdherent}
                                      updateAdherent={props.updateAdherent} >  
                                    </Adhrent>
                                    
                                   
                                    
                                    )) 
                                } 
                                
                                </tbody>
                                </table>
                                <br></br>
                                
                                
                            </div>
                            </div>
                           
                        </div>
                        </div>
  </section>       
         
        
     
    </div>
  );

  
}

