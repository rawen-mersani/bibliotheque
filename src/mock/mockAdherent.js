export default [
    {
        id:"1",
        name:"mersani",
        pname:"rawen",
        mail:"rawen@mail.com",
        tel:12345678,
        etat: "non",
        mdp: "rawen",
    },
    {
        id:"2",
        name:"missaoui",
        pname:"rym",
        mail:"mail2@mail.com",
        tel:12345678,
        etat: "non",
        mdp: "rym",
    },
  ]