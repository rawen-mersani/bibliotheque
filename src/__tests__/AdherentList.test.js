import React from "react"
import { render } from "@testing-library/react"
import AdherentList from "./../components/Adherent/AdherentList"
import mockAdherent from "../mock/mockAdherent"
import user from "@testing-library/user-event"

describe("test adherents list", () => {
  test("should render an array of tasks list", () => {
    const mockDeleteAdherent = jest.fn()
    const AdherentsComponent = render(
      <AdherentList adherents={mockAdherent}
      deleteAdherent={mockDeleteAdherent}  />
    )
    expect(AdherentsComponent).toMatchSnapshot()
    // debug()
  })

  test("test the integration of AdherentList and Adherent", () => {
    const mockDeleteAdherent = jest.fn()
    const mockAdherentsWithOneAdherent = [
      {
        id:"1",
        name:"mersani",
        pname:"rawen",
        mail:"rawen@mail.com",
        tel:12345678,
        etat: "non",
        mdp: "rawen",
        
      },
    ]
    const { getByTestId } = render(
      <AdherentList deleteAdherent={mockDeleteAdherent} adherents={mockAdherentsWithOneAdherent} />
    )
    const DeleteAdherentButton = getByTestId("deleteAdherent")
    user.click(DeleteAdherentButton)
    expect(mockDeleteAdherent).toHaveBeenCalled()
    expect(mockDeleteAdherent).toHaveBeenCalledTimes(1)
  })
})