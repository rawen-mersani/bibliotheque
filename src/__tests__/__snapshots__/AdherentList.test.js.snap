// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test adherents list should render an array of tasks list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <section
          class="content"
        >
          <div
            class="row"
          >
            <div
              class="col-xs-12"
            >
              <div
                class="box"
              >
                <div
                  class="box-header"
                >
                  <h3
                    class="box-title"
                  >
                    Liste adherents
                  </h3>
                  <br />
                  <br />
                  <button
                    class="btn btn-info"
                  >
                    Afficher liste
                  </button>
                </div>
                <div
                  class="box-body"
                >
                  <table
                    class="table table-bordered table-hover"
                    id="example2"
                  >
                    <thead>
                      <tr>
                        <th>
                          Nom
                        </th>
                        <th>
                          Prénom
                        </th>
                        <th>
                          Mail
                        </th>
                        <th>
                          Télephone
                        </th>
                        <th>
                          Etat
                        </th>
                        <th>
                          Détails
                        </th>
                        <th>
                          Bannir
                        </th>
                        <th>
                          Modifier
                        </th>
                        <th>
                          Supprimer
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td>
                          mersani
                        </td>
                        <td>
                          rawen
                        </td>
                        <td>
                          rawen@mail.com
                        </td>
                        <td>
                          12345678
                        </td>
                        <td>
                          non
                        </td>
                        <td>
                          <a
                            href="/Adhérents/1"
                          >
                            Afficher plus
                          </a>
                        </td>
                        <td>
                          <button
                            class="fa fa-ban btn btn-warning"
                          />
                        </td>
                        <td>
                          <button
                            class="fa fa-edit btn btn-info"
                          />
                        </td>
                        <td>
                          <button
                            class="fa fa-remove btn btn-danger"
                            data-testid="deleteAdherent"
                          />
                        </td>
                      </tr>
                      <tr>
                        <td>
                          missaoui
                        </td>
                        <td>
                          rym
                        </td>
                        <td>
                          mail2@mail.com
                        </td>
                        <td>
                          12345678
                        </td>
                        <td>
                          non
                        </td>
                        <td>
                          <a
                            href="/Adhérents/2"
                          >
                            Afficher plus
                          </a>
                        </td>
                        <td>
                          <button
                            class="fa fa-ban btn btn-warning"
                          />
                        </td>
                        <td>
                          <button
                            class="fa fa-edit btn btn-info"
                          />
                        </td>
                        <td>
                          <button
                            class="fa fa-remove btn btn-danger"
                            data-testid="deleteAdherent"
                          />
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <section
        class="content"
      >
        <div
          class="row"
        >
          <div
            class="col-xs-12"
          >
            <div
              class="box"
            >
              <div
                class="box-header"
              >
                <h3
                  class="box-title"
                >
                  Liste adherents
                </h3>
                <br />
                <br />
                <button
                  class="btn btn-info"
                >
                  Afficher liste
                </button>
              </div>
              <div
                class="box-body"
              >
                <table
                  class="table table-bordered table-hover"
                  id="example2"
                >
                  <thead>
                    <tr>
                      <th>
                        Nom
                      </th>
                      <th>
                        Prénom
                      </th>
                      <th>
                        Mail
                      </th>
                      <th>
                        Télephone
                      </th>
                      <th>
                        Etat
                      </th>
                      <th>
                        Détails
                      </th>
                      <th>
                        Bannir
                      </th>
                      <th>
                        Modifier
                      </th>
                      <th>
                        Supprimer
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        mersani
                      </td>
                      <td>
                        rawen
                      </td>
                      <td>
                        rawen@mail.com
                      </td>
                      <td>
                        12345678
                      </td>
                      <td>
                        non
                      </td>
                      <td>
                        <a
                          href="/Adhérents/1"
                        >
                          Afficher plus
                        </a>
                      </td>
                      <td>
                        <button
                          class="fa fa-ban btn btn-warning"
                        />
                      </td>
                      <td>
                        <button
                          class="fa fa-edit btn btn-info"
                        />
                      </td>
                      <td>
                        <button
                          class="fa fa-remove btn btn-danger"
                          data-testid="deleteAdherent"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td>
                        missaoui
                      </td>
                      <td>
                        rym
                      </td>
                      <td>
                        mail2@mail.com
                      </td>
                      <td>
                        12345678
                      </td>
                      <td>
                        non
                      </td>
                      <td>
                        <a
                          href="/Adhérents/2"
                        >
                          Afficher plus
                        </a>
                      </td>
                      <td>
                        <button
                          class="fa fa-ban btn btn-warning"
                        />
                      </td>
                      <td>
                        <button
                          class="fa fa-edit btn btn-info"
                        />
                      </td>
                      <td>
                        <button
                          class="fa fa-remove btn btn-danger"
                          data-testid="deleteAdherent"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
                <br />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
